pure_pursuit/src/pure_pursuit_core.cpp

```diff
@@ -146,6 +149,28 @@ void PurePursuitNode::run()
   }
 }
 
+void PurePursuitNode::publishControlCommands(
+  const bool& can_get_curvature, const double& kappa) const
+{
+  if (output_interface_ == "twist")
+  {
+    publishTwistStamped(can_get_curvature, kappa);
+  }
+  else if (output_interface_ == "ctrl_cmd")
+  {
+    publishCtrlCmdStamped(can_get_curvature, kappa);
+  }
+  else if (output_interface_ == "all")
+  {
+    publishTwistStamped(can_get_curvature, kappa);
+    publishCtrlCmdStamped(can_get_curvature, kappa);
+  }
+  else
+  {
+    ROS_WARN("[pure_pursuit] control command interface is not appropriate");
```

Comment:
Instead of failing here on every loop, the node should check the value of this paramter in the constructor and fail to launch with an error. This avoids the user assuming that things are working when they are not (especially if this node's output isn't going to `screen`).